hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/CsvMetadataResolver.java

```diff
@@ -42,7 +50,16 @@
     }
 
     @Override
-    protected SupplierEx<QueryTarget> queryTargetSupplier() {
-        return CsvQueryTarget::new;
+    protected SupplierEx<QueryTarget> queryTargetSupplier(List<MappingField> resolvedFields) {
+        List<String> fieldMap = createFieldList(resolvedFields);
+        return () -> new CsvQueryTarget(fieldMap);
+    }
+
+    @Nonnull
+    private static List<String> createFieldList(List<MappingField> resolvedFields) {
+        return resolvedFields.stream()
+                      .map(f -> f.externalName() != null ? f.externalName() : f.name())
+                      .distinct()
```

Comment:
Why `distinct` is needed here? Maybe it should be part of validation?

Reply:
You could theoretically add a mapping with two columns pointing to the same external name:
```
CREATE MAPPING my_file (
  col1 INT EXTERNAL NAME col1,
  col1_again INT EXTERNAL NAME col1
) ...
```
Will add a test for that.